components/CommonModalComponent.tsx

```diff
@@ -0,0 +1,351 @@
+import React, { useEffect, useMemo, useState } from "react";
+import { styled } from "styled-components";
+import { useRequest } from "../hooks/useRequest";
+import { FolderList } from "@/types";
+
+type props = {
+  show: boolean;
+  modalOff: () => void;
+  modalInfo: {
+    type: string;
+    title: string;
+    folderName: string;
+    linkName: string;
+  };
+};
+const CommonModalComponent = ({ show, modalOff, modalInfo }: props) => {
+  const { type, title, folderName, linkName } = modalInfo;
+
+  const { data: folderData, request } = useRequest<FolderList>({
+    url: "api/users/1/folders",
+    method: "GET",
+  });
+  const [selectLink, setSelectLink] = useState<number>(0);
```

Comment:
상태에 기본형 데이터를 저장하는 경우에는 타입을 잡아주지 않아도 괜찮아요!
물론 잡아주는것도 괜찮다고 생각합니다!
```suggestion
  const [selectLink, setSelectLink] = useState(0);
```